lib/kube/proxy/transport.go

```diff
@@ -379,6 +379,12 @@ func (f *Forwarder) remoteClusterDiater(clusterName string) dialContextFunc {
 		}
 
 		return targetCluster.DialTCP(reversetunnel.DialParams{
+			// Send a sentinel value to the remote cluster because this connection
+			// will be used to forward multiple requests to the remote cluster from
+			// different users.
+			// IP Pinning is based on the source IP address of the connection that
+			// we transport over HTTP headers so it's not affected.
+			From: &utils.NetAddr{AddrNetwork: "tcp", Addr: "0.0.0.0:0"},
```

Comment:
How can we protect against forgetting this again? Could `DialTCP` do a nil check and return an error or set the sentinel there?

Reply:
maybe a nil check wherever we were panicking?

Reply:
Added a test to protect that and it also validates the value provided.